checkmate/checker/pipeline/file.py

```diff
@@ -0,0 +1,67 @@
+"""Stages which read files."""
+
+import csv
+from logging import getLogger
+from zipfile import BadZipFile, ZipFile
+
+from checkmate.checker.pipeline.core import Stage
+from checkmate.exceptions import StageException
+
+LOG = getLogger(__name__)
+
```

Comment:
Atomic things you can do which can be remixed in any order (that connects up ok). These all read files and some spit out generators of things, and others spit out another file.

You can plug a file reading thing into a file writing thing etc.